modules/features/custom/application.go

```diff
@@ -132,3 +134,296 @@ func init() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+
+// GetDNS returns the DNS layer of the packet or nil
+func GetDNS(new interface{}) *layers.DNS {
+	tl := new.(packet.Buffer).TransportLayer()
+	if tl != nil {
+		payload := tl.LayerPayload()
+		if tl.LayerType() == layers.LayerTypeTCP {
```

Comment:
I did this 2 years ago, so my memory might fail. But it seems to me it's also doing UDP, right?
Just that on TCP packets, the beginning of the payload is ignored.

Yeah I'm pretty sure this assumes the TCP connection is very nice and pretty, and ignores edge cases.
When I was doing this, it worked well enough for my dataset.

I still made the PR though, since a half working thing is still better than no working thing :) (and IIRC the half-working thing is only in this `GetDNS` function, isolated from the feature functions, for easy fixing in the future)
Unfortunately I don't think I have the time to do this properly right now. I was hoping someone else who needed it could do it later.